types/detect/detect.go

```diff
@@ -67,3 +68,10 @@ type Logger interface {
 	Warnw(format string, v ...interface{})
 	Errorw(format string, v ...interface{})
 }
+
+type DataSource interface {
+	Get(string) (interface{}, bool)
+	ValueType() string
+	Namespace() string
```

Comment:
Namespace is to avoid ID collisions when we allow external users to plug their own data sources from plugins (in CNDR we already have such a usecase).
ID is basically a data source name.
They practically make a joint key for a DataSource.

Reply:
So can you put it as a comment on each of them and maybe on the interface so it will be clear?

Reply:
Hum, I left a comment saying that the map[string]map[string]XX line should also mention it (sorry for the duplicate request).